src/lib.rs

```diff
@@ -349,53 +349,48 @@ pub enum Error {
     InvalidParityValue(key::InvalidParityValue),
 }
 
-impl Error {
-    fn as_str(&self) -> &str {
-        match *self {
-            Error::IncorrectSignature => "secp: signature failed verification",
-            Error::InvalidMessage => "secp: message was not 32 bytes (do you need to hash?)",
-            Error::InvalidPublicKey => "secp: malformed public key",
-            Error::InvalidSignature => "secp: malformed signature",
-            Error::InvalidSecretKey => "secp: malformed or out-of-range secret key",
-            Error::InvalidSharedSecret => "secp: malformed or out-of-range shared secret",
-            Error::InvalidRecoveryId => "secp: bad recovery id",
-            Error::InvalidTweak => "secp: bad tweak",
-            Error::NotEnoughMemory => "secp: not enough memory allocated",
-            Error::InvalidPublicKeySum => "secp: the sum of public keys was invalid or the input vector lengths was less than 1",
-            Error::InvalidParityValue(_) => "couldn't create parity",
-        }
-    }
-}
-
-// Passthrough Debug to Display, since errors should be user-visible.
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        f.write_str(self.as_str())
+        use Error::*;
+
+        f.write_str(match *self {
+            IncorrectSignature => "secp: signature failed verification",
+            InvalidMessage => "secp: message was not 32 bytes (do you need to hash?)",
+            InvalidPublicKey => "secp: malformed public key",
+            InvalidSignature => "secp: malformed signature",
+            InvalidSecretKey => "secp: malformed or out-of-range secret key",
+            InvalidSharedSecret => "secp: malformed or out-of-range shared secret",
+            InvalidRecoveryId => "secp: bad recovery id",
+            InvalidTweak => "secp: bad tweak",
+            NotEnoughMemory => "secp: not enough memory allocated",
+            InvalidPublicKeySum => "secp: the sum of public keys was invalid or the input vector lengths was less than 1",
+            InvalidParityValue(_) => "couldn't create parity",
+        })
```

Comment:
I believe it was previously more readable, but we should use `write_err!()` anyway. I think it's OK to copy-paste the macro?

Reply:
Done, as suggested. Cheers.